src/main/scala/com.gu.gibbons/UserReminder.scala

```diff
@@ -28,28 +28,22 @@ class UserReminder[F[_] : Monad](settings: Settings, email: EmailService[F], bon
       * 3- Sends a reminder email to each user 
       * 4- Update keys to log when a reminder has been sent
       */
-    def run(now: OffsetDateTime, dryRun: Boolean): F[Result] = {
-      val fromBefore = now.minus(Settings.inactivityPeriod).toInstant
-      
-      if (dryRun) 
-        for {
-          _ <- logger.info("Yop, who's up for receiving a reminder?")
-          users <- bonobo.getUsers(fromBefore)
-        } yield DryRun(users)
-      else
-        for {
-          _ <- logger.info(s"Getting all the users older than ${Settings.inactivityPeriod}")
-          users <- bonobo.getUsers(fromBefore)
-          _ <- logger.info(s"Found ${users.length} users. Let's send some emails...")
-          nowI = now.toInstant
-          ress <- users.filterNot(u => Settings.whitelist(u.id.id)).traverse { user => 
-            for {
-              newUser <- bonobo.setRemindedOn(user, nowI)
-              res <- email.sendReminder(newUser)
-            } yield (newUser.id -> res)
-          }.map(_.toMap)
-          _ <- logger.info("aaaand that's a wrap! See you next time.")
-        } yield FullRun(ress)
-    }
+    def run(now: OffsetDateTime, dryRun: Boolean): F[Map[UserId, EmailResult]] =
+      for {
+        _ <- logger.info(s"Getting all the users older than ${Settings.inactivityPeriod}")
+        users <- bonobo.getUsers(now.minus(Settings.inactivityPeriod).toInstant)
+        _ <- logger.info(s"Got ${users.length}... but we only need developer accounts")
+        devs <- users.foldMapM(u => bonobo.isDeveloper(u).map(b => if (b) Vector(u) else Vector.empty))
```

Comment:
Due to the way `isDeveloper` is implemented, this line suffers from the [`N+1` Query problem](https://stackoverflow.com/questions/97197/what-is-the-n1-select-query-issue). Does DynamoDB support a query with multiple IDs (e.g. using `IN` or multiple `OR`s)?

Reply:
There is, but I need to raise a PR to add that capability in Scanamo. Good spot 👍